neicnordic/sensitive-data-archive
# Feature/api keys

**Related issue(s) and PR(s)**  
This PR closes #1036.


**Description**
With this PR an endpoint for adding key hash in added in the api service.
An extra step of adding an api role in the database was necessary in order to make it work.

**Comment**
There is no integration test for this feature because Malin has already created one and it will be [in this PR](https://github.com/neicnordic/sensitive-data-archive/pull/1073) which is on hold because of the current.

**How to test**
From the root of the folder run `make build-all`
and then for linux run:
```bash
PR_NUMBER=$(/usr/bin/date '+%F')  docker compose -f .github/integration/sda-s3-integration.yml run integration_test
```
while for mac run:
```bash
PR_NUMBER=$(/bin/date '+%F')  docker compose -f .github/integration/sda-s3-integration.yml run integration_test
```

Then set the token as envvar: `export ACCESS_TOKEN=$(curl -s -k http://localhost:8080/tokens | jq -r '.[0]')`
and run:
```bash
curl -H "Authorization: Bearer $ACCESS_TOKEN" -H "Content-Type: application/json" -X POST -d '{"hash": "this-is-just-a-hash", "description": "this is keys description"}' http://localhost:8090/key/hashed
```

Confirm that the key hash is in the database by running: `docker exec -it postgres psql -U postgres -d sda` and then the query `select * from sda.encryption_keys;`
